samples/net/wpanusb/boards/atsamr21_xpro.conf

```diff
@@ -0,0 +1,7 @@
+#
+# Copyright (c) 2019, Gerson Fernando Budke
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+CONFIG_NET_CONFIG_IEEE802154_DEV_NAME="RF2XX_0"
```

Comment:
We do not have the configs for other boards, why we need it for SAMR21?

Reply:
I don't know if is really need but seems to be the right thing to do.

I'll add context here, there are two things: 1- SAMR21 is a SiP that incorporate at86rf233; 2- the at86rf23x uses RF2xx driver, which allows more than one iface at same time. That said, there will be cases with 2 transceivers at same board and user need select one (2.4G or Sub-Giga)? In my perspective, user should use same approach always to select the desired option but I know not always is it this possible.

There are others examples that were defined a board conf. For an MCU using external transceiver user always will need an overlay or a specific board config and an overlay will not handle all cases for rf2xx, but could help with right documentation.

I'm thinking about it and maybe an overlay file for rf2xx driver will be better. This will fix first iface for all boards using rf2xx driver. The documentation could explain how to build for second iface, switching from _RF2XX_0_ to _RF2XX_1_.

I would like know what happen if we add **CONFIG_NET_CONFIG_IEEE802154_DEV_NAME="RF2XX_0** to soc config and than a user uses an overlay that point to **RF2XX_1** for mistake with a atsamr21_xpro board for instance. This case could be on another example or application and I would like to have consistence to avoid user frustration with driver and/or zephyr.

What do you think?

Reply:
I think it should be solved in the same way as for [NRF52833](https://github.com/zephyrproject-rtos/zephyr/blob/master/soc/arm/nordic_nrf/nrf52/Kconfig.defconfig.nrf52833_QIAA).

Reply:
@jfischer-phytec-iot Could you take time to check #21468 before I agreed with your opinion? This is one example that I'll submit soon. Your opinion is important to me understand that situation. At this time you can complete ignore _Shippable_.

Reply:
We probably need something similar to ETH_NET_DEVICE_INIT, also IEEE802154_NET_DEVICE_INIT.

Reply:
Well, I submitted the proposal at #21560 to address the issue.